### PR TITLE
chore: minor clean-ups

### PR DESCRIPTION
**Pull request by bartoszmajsak**
## Description

While working on a Service Mesh component, I discovered several areas in the codebase that could be slightly improved. This pull request aims to address those issues for better code quality and maintainability.

### Changes

- Improved error handling by properly addressing previously ignored errors
- Unified receiver names across funcs for all components
- Updated deprecated methods such as `AddToScheme` and `ioutil` functions
- Cleaned up the code by removing unneeded string conversions, redundant import aliases, and unnecessary variables in loops.
- Improved error handling specifically when checking managed RHODS.
- Wrapped existing errors rather than constructing new ones, for more context in error messages.
- Fixed documentation comments to follow GoDoc conventions, including starting comments with the function name.
- Removed an unused module and dead code
- Fixed typos

These changes are proposed separately from the component work, so the final PR won't be that large, but also to speed up their review and potential inclusion into the main codebase.

## How Has This Been Tested?

`make test`

## Merge criteria:
- [ ] The commits are squashed in a cohesive manner and have meaningful messages.
- [ ] Testing instructions have been added in the PR body (for PRs involving changes that are not immediately obvious).
- [x] The developer has manually tested the changes and verified that the changes work

### CODE REVIEW

File: pkg/deploy/deploy.go [hunk 1]
```diff
@@ -384,9 +389,9 @@ func SubscriptionExists(cli client.Client, namespace string, name string) (bool,
 	return true, nil
 }
 
-// Checks if an Operator with operatorprefix is installed in the given namespace list
-// TODO: if we need to check exact verison of the operator installed, can append vX.Y.Z later
+// OperatorExists checks if an Operator with 'operatorprefix' is installed in the given namespace.
```
Comment (zdtsw) [inline pkg/deploy/deploy.go:392]:
```suggestion
// OperatorExists checks if an Operator with 'operatorprefix' is installed.
```
my previous comment was wrong, it does not require a given namespace.

File: components/dashboard/dashboard.go [hunk 2]
```diff
@@ -209,7 +210,7 @@ func (d *Dashboard) ReconcileComponent(cli client.Client, owner metav1.Object, d
 			"<section-title>":       "OpenShift Managed Services",
 		})
 		if err != nil {
-			return fmt.Errorf("Error replacing with correct dashboard url for ConsoleLink: %v", err)
+			return errors.Wrap(err, "failed replacing with correct dashboard url for ConsoleLink: %v")
```
Comment (zdtsw) [inline components/dashboard/dashboard.go:213]: do you want to change all the components with this or only dashbaord?

File: pkg/deploy/deploy.go (hunk 1 above)
Comment (bartoszmajsak) [reply pkg/deploy/deploy.go:392]: You can ```suggest and we can get it fixed as part of this pr

File: components/dashboard/dashboard.go (hunk 2 above)
Comment (bartoszmajsak) [reply components/dashboard/dashboard.go:213]: Good hint. I'll check the others.

File: pkg/deploy/deploy.go (hunk 1 above)
Comment (zdtsw) [reply pkg/deploy/deploy.go:392]: haha thought i made a suggestion, by removing "in the given namespace."

File: go.mod [hunk 3]
```diff
@@ -95,7 +95,6 @@ require (
 
 replace (
 	//github.com/go-openapi/spec => github.com/go-openapi/spec v0.19.3
-	github.com/tektoncd/pipeline => github.com/tektoncd/pipeline v0.12.0
```
Comment (zdtsw) [inline go.mod:98]: @VaishnaviHire remember any reason we did these here?

Comment (VaishnaviHire) [reply go.mod:98]: I think this was required for one of the dependency. But we we have a clean `go mod` , it is fine to remove this

**Comment by VaishnaviHire** [comment]
Thanks @bartoszmajsak for these fixes! Changes look good to me

/lgtm

**Review by VaishnaviHire — APPROVED** [review]

File: components/dashboard/dashboard.go (hunk 2 above)
Comment (bartoszmajsak) [reply components/dashboard/dashboard.go]: @zdtsw this is now reworked using `fmt.Errorf` and `%w` directive, which essentially does the same.

**Comment by bartoszmajsak** [comment]
@VaishnaviHire thanks, I added one additional thing while cleaning up error handling, please take a look when you'll find some time [`cdc4416` (#536)](https://github.com/opendatahub-io/opendatahub-operator/pull/536/commits/cdc4416bde0fae39da7541096317addd42236e65)

**Review by zdtsw — APPROVED** [review]
/lgtm

**Pull request merged**